src/transactions/transaction_signed.rs

```diff
@@ -1,22 +1,38 @@
-use crate::sf::ethereum::r#type::v2::TransactionTrace;
 use crate::transactions::error::TransactionError;
-use reth_primitives::{Signature, Transaction, TransactionSigned};
+use reth_primitives::TransactionSigned;
 use revm_primitives::{hex, B256};
+use sf_protos::ethereum::r#type::v2::TransactionTrace;
 use std::str::FromStr;
 
-impl TryFrom<&TransactionTrace> for TransactionSigned {
-    type Error = TransactionError;
+use super::{signature::signature_from_trace, transaction::trace_to_transaction};
 
-    fn try_from(trace: &TransactionTrace) -> Result<Self, Self::Error> {
-        let transaction = Transaction::try_from(trace)?;
-        let signature = Signature::try_from(trace)?;
-        let hash = B256::from_str(&hex::encode(trace.hash.as_slice()))
-            .map_err(|_| TransactionError::MissingCall)?;
-        let tx_signed = TransactionSigned {
-            transaction,
-            signature,
-            hash,
-        };
-        Ok(tx_signed)
-    }
+pub fn trace_to_signed(trace: &TransactionTrace) -> Result<TransactionSigned, TransactionError> {
+    let transaction = trace_to_transaction(trace)?;
+    let signature = signature_from_trace(trace)?;
+    let hash = B256::from_str(&hex::encode(trace.hash.as_slice()))
+        .map_err(|_| TransactionError::MissingCall)?;
+    let tx_signed = TransactionSigned {
+        transaction,
+        signature,
+        hash,
+    };
+    Ok(tx_signed)
 }
+
+// impl TryFrom<&TransactionTrace> for TransactionSigned {
+// type Error = TransactionError;
+//
+// fn try_from(trace: &TransactionTrace) -> Result<Self, Self::Error> {
+// let transaction = Transaction::try_from(trace)?;
+// let signature = Signature::try_from(trace)?;
+// let hash = B256::from_str(&hex::encode(trace.hash.as_slice()))
+// .map_err(|_| TransactionError::MissingCall)?;
+// let tx_signed = TransactionSigned {
+// transaction,
+// signature,
+// hash,
+// };
+// Ok(tx_signed)
+// }
+// }
+//
```

Comment:
```suggestion
```

Remove commented code

Reply:
Another small comment, make sf protos repo public, so that CI CD can run correctly here